app/src/app/leaderboards.tsx

```diff
@@ -65,25 +92,22 @@ export default function Leaderboards() {
         switch (leaderboardType) {
             case LeaderboardTypes.TOTAL_WINS_PER_TEAM:
                 setLeaderboardAbbrev(LeaderboardAbbrevs.TOTAL_WINS_PER_TEAM);
+                setNumPages(numTeamPages); 
                 break;
             case LeaderboardTypes.AVERAGE_POINTS_PER_PLAYER:
                 setLeaderboardAbbrev(LeaderboardAbbrevs.AVERAGE_POINTS_PER_PLAYER);
+                setNumPages(numPlyrPages); 
                 break;
             case LeaderboardTypes.MOST_BOOKMARKED_PLAYERS:
                 setLeaderboardAbbrev(LeaderboardAbbrevs.MOST_BOOKMARKED_PLAYERS);
+                setNumPages(numPlyrPages); 
                 break;
             case LeaderboardTypes.PERCENTAGE_WINS_PER_TEAM:
                 setLeaderboardAbbrev(LeaderboardAbbrevs.PERCENTAGE_WINS_PER_TEAM);
+                setNumPages(numTeamPages); 
                 break;
         }
     }, [leaderboardType]);
```

Comment:
eslint gives error here, please add `// eslint-disable-next-line react-hooks/exhaustive-deps` before this line